creator-node/src/snapbackSM/syncHistoryAggregator.js

```diff
@@ -0,0 +1,133 @@
+const moment = require('moment')
+
+const redisClient = require('../redis')
+const { logger: genericLogger } = require('../logging')
+const config = require('../config')
+
+const CREATOR_NODE_ENDPOINT = config.get('creatorNodeEndpoint')
+
+const SYNC_STATES = Object.freeze({
+  triggered: 'triggered',
+  success: 'success',
+  fail: 'fail'
+})
+
+// Make key expire in 7 days in seconds
+const EXPIRATION = 7 /* days */ * 24 /* hr */ * 60 /* min */ * 60 /* s */
+
+/**
+ * Class intended to:
+ * - Record the number of the current day's triggered, successful, and failed sync attempts
+ * - Record the timestamp of the most recent successful and failed sync
+ */
+
+// Note: When recording a sync 'success' or 'fail', it implies that the secondary was able to respectfully
+// successfully or unsuccessfully take the primary's exported data, and update its own state
+
+class SyncHistoryAggregator {
+  static async recordSyncSuccess (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.success,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncFail (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.fail,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncData ({ state, timeOfEvent, logContext }) {
+    const logger = genericLogger.child(logContext)
+
+    try {
+      // Update aggregate sync data
+      const aggregateSyncKeys = SyncHistoryAggregator.getAggregateSyncKeys(CREATOR_NODE_ENDPOINT)
+
+      // Get the TTL if the key exists for the number of triggered syncs, and update the value by 1
+      const aggregateSyncKeyTriggeredTTL = await this.getKeyTTL(aggregateSyncKeys.triggered)
```

Comment:
in this case counting triggered seems unnecessary since that would just be success + failed. triggered would be useful if we were counting triggered first and then count success/fail at the end but i don't think thats necessary